.github/workflows/cypress.yml

```diff
@@ -0,0 +1,88 @@
+name: Cypress Tests
+
+on:
+  push:
+    branches:
+      - '**'
+  # Allows workflow to be called from other workflows
+  workflow_call:
+    inputs:
+      ref:
+        required: true
+        type: string
+
+# Avoid duplicate workflows on same branch
+concurrency: 
+  group: ${{ github.workflow }}-${{ github.ref }}-cypress
+  cancel-in-progress: true
+
+jobs:
+  e2e_tests:
+    runs-on: ubuntu-latest
+
+    defaults:
+      run:
+        shell: bash -ileo pipefail {0}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        specs: [ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21 ]
+    
+    steps:
+      - name: Checkout Streamlit code
+        uses: actions/checkout@v3
+        with: 
+          ref: ${{ inputs.ref }}
+          persist-credentials: false
+          submodules: 'recursive'
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v4
+        with:
+            python-version: "3.10"
+      - name: Setup virtual env
+        uses: ./.github/actions/make_init
+      - name: Run make develop
+        run: make develop
+      - name: Register Streamlit User & Mapbox Token
+        run: |
+          if [ ! -d $HOME/.streamlit ] ; then
+            mkdir $HOME/.streamlit
+          fi
+          echo '[mapbox]' >  ~/.streamlit/config.toml
+          MAPBOX_TOKEN=$(curl -sS https://data.streamlit.io/tokens.json | jq -r '.["mapbox-localhost"]') \
+            && echo 'token = "'$MAPBOX_TOKEN'"' >> ~/.streamlit/config.toml
+          
+          echo '[general]' >  ~/.streamlit/credentials.toml
+          echo 'email = "test@streamlit.io"' >> ~/.streamlit/credentials.toml
+      - name: Install Cypress dependencies
+        run: |
+          sudo apt install -y xvfb libgtk2.0-0 libnotify-dev libgconf-2-4 libnss3 libxss1 libasound2 jq curl
```

Comment:
nits:

* you can probably drop the `jq` install since it's used in the command above, so I think it already has to be installed at this point
* I'd move this to be the first step run even if it technically doesn't have to happen until now since installing dependencies earlier on feels more natural